File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> umap(256, 0);
        int res = 0;
        int left = 0, right = 0;

        for(right = 0; right < s.size(); right++){
            if(umap[s[right]] == 0){
                umap[s[right]] = 1;
            }
            else{
                while(umap[s[right]] != 0){
                    umap[s[left]]--;
                    left++;
                }
                umap[s[right]] = 1;
            }
            res = max(res, right - left + 1);
        }

        return res;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        balanced = True

        def helper(node):
            nonlocal balanced
            if not node:
                balanced = balanced and True
                return 0
            l, r = helper(node.left), helper(node.right)
            balanced = balanced and (abs(l - r) <= 1)
            return 1 + max(l, r)

        helper(root)        
        return balanced
        